warmup/bootstrapper.py

```diff
@@ -0,0 +1,76 @@
+#!/bin/env pypy
+
+"""
+This script is designed to be run with PyPy via a pipe.
+
+It will read JSON format data from STDIN, and will write a comma-separated pair
+of floats (mean, CI) on STDOUT.
+
+Input data should be as per. the needs of the tables for the "main" warmup
+experiment -- i.e. a list of pexecs, each containing a list of (steady state)
+segments, each containing a list of floats.
+
+Much of the code here comes from libkalibera.
+"""
+
+import math
+import random
+
+from decimal import Decimal, ROUND_UP, ROUND_DOWN
+
+
+BOOTSTRAP_ITERATIONS = 100000
+CONFIDENCE_LEVEL = '0.99'  # Must be a string to pass to Decimal.
+
+
+def _mean(data):
+    return math.fsum(data) / float(len(data))
+
+
+def bootstrap_steady_perf(steady_segments_all_pexecs, confidence_level=CONFIDENCE_LEVEL):
+    """This is not a general bootstrapping functions.
+    Input is a list containing a list for each pexec, containing a list of
+    segments with iteration times.
+    """
+
+    # How many bootstrap samples do we need from each pexec? We want at least
+    # BOOTSTRAP_ITERATIONS samples over all.
+    n_resamples = int(math.floor(BOOTSTRAP_ITERATIONS / len(steady_segments_all_pexecs))) + 1
+    bootstrapped_samples = list()  # Final list of BOOTSTRAP_ITERATIONS resamples.
+
+    for segments in steady_segments_all_pexecs:  # Iterate over pexecs.
+        for _ in xrange(n_resamples):
+            sample = list()
+            for seg in segments:
+                sample.extend([random.choice(seg) for _ in xrange(len(seg))])
+            bootstrapped_samples.append(sample)
+    assert len(bootstrapped_samples) >= BOOTSTRAP_ITERATIONS
```

Comment:
The current comment doesn't explain this sufficiently -- at least not as well as your GH comment does!